cmake/cmake_test/add_dir.cmake

```diff
@@ -130,6 +130,7 @@ function(ct_add_dir _ad_test_dir)
         )
 
         # Configure the CMakeLists.txt for test in the build directory
+        cpp_enabled_languages(_CT_LANGUAGES)
```

Comment:
It may be useful to allow the enabled languages for the tests be configurable through something like a `LANGUAGES` kwarg on `ct_add_dir()`. Then, here you could check if `ADD_DIR_LANGUAGES` is empty to decide if you will query the enabled languages or just set it to `ADD_DIR_LANGUAGES`.

Reply:
I like this idea! We could default it to C to restore the old behavior.